internal/dcache/cluster_manager/cluster_manager_impl.go

```diff
@@ -508,7 +510,155 @@ func (cmi *ClusterManagerImpl) updateStorageClusterMapIfRequired() {
 
 }
 
-func (cmi *ClusterManagerImpl) updateMVList(clusterMapRVMap map[string]dcache.RawVolume, clusterMapMVMap map[string]dcache.MirroredVolume) {
+func (cmi *ClusterManagerImpl) updateMVList(rvMap map[string]dcache.RawVolume, existingMVMap map[string]dcache.MirroredVolume, NumReplicas int, MvsPerRv int) map[string]dcache.MirroredVolume {
+
+	// Local types
+	type rv struct {
+		rvName string
+		slots  int //MvsPerRv
+	}
+
+	type node struct {
+		nodeId string
+		rvs    []rv
+	}
+
+	nodeToRvs := make(map[string]node)
+
+	// TODO :: Handle scenarios for degraded and fix scenarios
+	// Degraded - When any of the rv's in a mv is offline make mv as degraded [Done]
+	// Fix - Replace any rv which is offline with an available rv and mark rv as out-of-sync while mv's state will be degraded
+	// Sync - This will be handled by replica manager and it will change mv state to syncing
+	// Offline - Mark a mv as offline when all the rv's within it are offline [Done]
+
+	// Populate the RV struct and node struct
+	for rvId, rvInfo := range rvMap {
+		if rvInfo.State == dcache.StateOffline {
+			// Skip RVs that are offline
```

Comment:
usually comments that say why there are doing something are more useful than comments that say what they are doing.
so

// Skip offline RVs as they cannot contribute to any MV.

Reply:
also let's assert that RV state can only be Offline or Online